server/src/main/java/org/elasticsearch/index/query/CoordinatorRewriteContextProvider.java

```diff
@@ -49,18 +51,26 @@ public CoordinatorRewriteContext getCoordinatorRewriteContext(Index index) {
         if (indexMetadata == null) {
             return null;
         }
-        DateFieldMapper.DateFieldType dateFieldType = mappingSupplier.apply(index);
-        if (dateFieldType == null) {
+        Map<String, DateFieldMapper.DateFieldType> dateFieldMap = mappingSupplier.apply(index);
+        if (dateFieldMap == null) {
             return null;
         }
+
+        DateFieldMapper.DateFieldType timestampFieldType = dateFieldMap.get(DataStream.TIMESTAMP_FIELD_NAME);
         IndexLongFieldRange timestampRange = indexMetadata.getTimestampRange();
-        if (timestampRange.containsAllShardRanges() == false) {
-            timestampRange = indexMetadata.getTimeSeriesTimestampRange(dateFieldType);
+        if (timestampRange.containsAllShardRanges() == false) { /// MP TODO: is this logic needed for event.ingested? Not clear what this is
```

Comment:
@martijnvg - Could you help me understand what `timestampRange.containsAllShardRanges()` is doing and why it being false means we need to check `getTimeSeriesTimestampRange()`?

And does that check have any bearing on the event.ingested timestamp range?

Reply:
It has been a while since I looked at this. But the background is that for the `@timestamp` we do additional filtering of shards. 

Each tsdb backing index has an `index.time_series.start_time` and `index.time_series.end_time` index settings. Each document has `@timestamp` field. During indexing documents are routed to the backing index that matches with the start end and time range.

For searches with a range query on `@timestamp` field, this gives us the ability to filter out shards that have backing indices that do not match with the range.

This logic here is a fallback to use the time serie range (based on start and end time index settings in cluster state) in case the range matches with shards that are not read only (I guess not yet frozen or backed by searchable snapshot). Obviously this only applies to tsdb and the `@timestamp` field. So I think this doesn't apply to the `event.ingested` field.